app/logic/users.py

```diff
@@ -10,28 +12,79 @@ def isEligibleForProgram(program, user):
     :param user: accepts a User object or userid
     :return: True if the user is not banned and meets the requirements, and False otherwise
     """
-
-    if (ProgramBan.select().where(ProgramBan.user == user, ProgramBan.program == program).exists()):
+    now = datetime.datetime.now()
+    if (ProgramBan.select().where(ProgramBan.user == user, ProgramBan.program == program, ProgramBan.endDate > now).exists()):
         return False
 
     return True
 
-def addRemoveInterest(rule, program_id, currentUser):
+def addUserInterest(program_id, username):
+    """
+    This function is used to add an interest to .
+    Parameters:
+    program_id: id of the program the user is interested in
+    username: username of the user showing interest
+    """
+    try:
+        Interest.get_or_create(program = program_id, user = username)
+        return "Successfully added interest"
+    except Exception as e:
+        print("Error: ", e)
+        return "An error ocurred while adding the interest"
```

Comment:
As I mentioned to you in person, blanket exceptions can be risky. If you expect a certain exception, catch it specifically. If your function returns errors, the calling code should handle that. If your function raises exceptions, the calling code should handle that. Mixing exceptions and return values can get tricky.

Reply:
Fixed.